checks.d/mongo.py

```diff
@@ -326,3 +391,29 @@ def check(self, instance):
             if m in self.RATES:
                 m = self.normalize(m.lower(), 'mongodb') + "ps"
                 self.rate(m, value, tags=tags)
+
+        stat_metrics = filter(lambda x: x.startswith('stats.'), self.METRICS)
+        for s in dbstats:
+            for m in stat_metrics:
+                value = dbstats[s]
+                dbtag = s  # key is db name, so we use that as the tag
+                try:
+                    for c in m.split('.'):
+                        value = value[c]
```

Comment:
if i understand correctly, c is either `stats` or the metric name. If that is the case you can simply do `value = value[m.split('.')[1]]` and drop the try/except.